picoquicfirst/picoquicdemo.c

```diff
@@ -276,10 +276,52 @@ typedef struct st_client_loop_cb_t {
     struct sockaddr_storage server_address;
     struct sockaddr_storage client_address;
     struct sockaddr_storage client_alt_address;
+    int client_alt_if;
     picoquic_connection_id_t server_cid_before_migration;
     picoquic_connection_id_t client_cid_before_migration;
 } client_loop_cb_t;
 
+#ifdef _WINDOWS
+char *strsep(char **stringp, const char *delim) {
+    if (*stringp == NULL) {
+        return NULL;
+    }
+    char *token_start = *stringp;
+    *stringp = strpbrk(token_start, delim);
+    if (*stringp) {
+        **stringp = '\0';
+        (*stringp)++;
+    }
+    return token_start;
+}
+#endif
+
```

Comment:
I understand your concern about "writing strsep for windows", but i think it is one of those cases in which we have to write our own code. My only concern with your implementation is that you reuse the name "strsep", which may cause problems if some other implementer creates their own copy of strsep in a different module, or in their application. I would prefer something like:

```
char * picoquic_strsep(char **stringp, const char *delim)
{
#ifdef _WINDOWS
    if (*stringp == NULL) {
        return NULL;
    }
    char *token_start = *stringp;
    *stringp = strpbrk(token_start, delim);
    if (*stringp) {
        **stringp = '\0';
        (*stringp)++;
    }
    return token_start;
#else 
    return strsep(stringp, delim);
#endif
}
```

and then use `picoquic_strsep` in your parsing code.